CarRentals/CarRentalsWebAPI/Controllers/BrandsController.cs

```diff
@@ -0,0 +1,86 @@
+﻿using CarRentalsWebAPI.DTO;
+using CarRentalsWebAPI.Models;
+using CarRentalsWebAPI.Services;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+
+namespace CarRentalsWebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BrandsController : ControllerBase
+    {
+        private readonly IBrandService _brandService;
+
+        public BrandsController(IBrandService brandService)
+        {
+            _brandService = brandService;
+        }
+
+        // GET: api/Brands
+        [HttpGet]
+        public ActionResult<List<BrandDto>> GetBrands()
+        {
+            List<BrandDto> list = new List<BrandDto>();
+
+            foreach (Brand item in _brandService.GetAll())
+            {
+                list.Add(new BrandDto(item));
+            }
+
+            return Ok(list);
+        }
+
+        // GET: api/Brands/5
+        [HttpGet("{id}")]
+        public ActionResult<BrandDto> GetBrand(int id)
+        {
+            var brand = _brandService.Get(id);
+
+            if (brand == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(new BrandDto(brand));
+        }
+
+        // PUT: api/Brands/5
+        [HttpPut("{id}")]
+        public IActionResult PutBrand(int id, BrandDto brand)
+        {
+            var toUpdate = _brandService.Update(id, BrandDto.DtoToEntity(brand));
+
+            if (toUpdate == null)
+            {
+                return NotFound();
+            }
+
+            return NoContent();
+        }
+
+        // POST: api/Brands
+        [HttpPost]
+        public ActionResult<BrandDto> PostBrand(BrandDto brandDto)
```

Comment:
there's a thing between ActionResult<T> and IActionResult and when to use each one. Check https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-5.0#iactionresult-type and let me know if it's still not clear and we can discuss